specs/pkg-js/trustless-pathing/testcase.js

```diff
@@ -0,0 +1,89 @@
+import { CID } from 'multiformats/cid'
+
+class TestCase {
+    constructor(name, root, path, scope, duplicates, byteRange, expectedCids) {
+        this.name = name
+        this.root = root
+        this.path = path
+        this.scope = scope
+        this.duplicates = duplicates
+        this.byteRange = byteRange
+        this.expectedCids = expectedCids
+    }
+
+    asQuery() {
+        let pp = this.path.split('/').filter(Boolean).join('/')
+        if (pp !== '') {
+            pp = '/' + pp
+        }
+        let br = ''
+        if (this.byteRange !== '') {
```

Comment:
Will be string or `null` after get from `URLSearchParams`.

```suggestion
        if (this.byteRange) {
```